packages/grid/_modules_/grid/hooks/features/useComponents.tsx

```diff
@@ -68,18 +61,40 @@ export const useComponents = (
 
   const paginationComponent = React.useMemo(
     () =>
-      componentOverrides?.pagination
-        ? React.createElement(componentOverrides.pagination, componentProps)
-        : null,
+      componentOverrides?.Pagination ? (
+        React.createElement(componentOverrides.Pagination, componentProps)
+      ) : (
+        <Pagination />
+      ),
     [componentOverrides, componentProps],
   );
 
+  const footerComponent = componentOverrides?.Footer ? (
+    React.createElement(componentOverrides.Footer, componentProps)
+  ) : (
+    <DefaultFooter paginationComponent={paginationComponent} />
+  );
+
   const renderError = React.useCallback(
     (props) => {
-      const ErrorOverlay = componentOverrides?.errorOverlay || ErrorMessage;
+      const ErrorOverlay = componentOverrides?.ErrorOverlay || ErrorMessage;
       return <ErrorOverlay {...componentProps} {...props} />;
     },
-    [componentOverrides?.errorOverlay, componentProps],
+    [componentOverrides?.ErrorOverlay, componentProps],
+  );
+
+  const renderColumnMenu = React.useCallback(
+    (props: GridColumnHeaderMenuItemProps) => {
+      return componentOverrides?.ColumnMenu ? (
+        React.createElement(componentOverrides.ColumnMenu, { ...componentProps, ...props })
+      ) : (
+        <DefaultGridColumnHeaderMenuItems
+          hideMenu={props.hideMenu}
+          currentColumn={props.currentColumn}
+        />
+      );
+    },
+    [componentOverrides, componentProps],
   );
```

Comment:
```suggestion
const {
  ColumnMenu = DefaultGridColumnHeaderMenuItems;
} = componentOverrides || {};

const renderColumnMenu = React.useCallback(
  (props: GridColumnHeaderMenuItemProps) => <ColumnMenu {...props} />,
  [ColumnMenu],
);
```

Reply:
Also, notice how we handle the components in the core package: https://github.com/mui-org/material-ui/blob/93479e811f88bd3dd0c4ff6ccb5c9bf01c29d0c3/packages/material-ui-unstyled/src/SliderUnstyled/SliderUnstyled.js#L551 or in the lab package:

Reply:
Do we need the `useComponents` indirection?